contracts/access/GatewayAccessControl.sol

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+bytes32 constant OWNER = keccak256("OWNER");
+bytes32 constant LOW_LEVEL_OPERATOR = keccak256("LOW_LEVEL_OPERATOR");
+bytes32 constant HIGH_LEVEL_OPERATOR = keccak256("HIGH_LEVEL_OPERATOR");
+bytes32 constant FINANCIAL_OWNER = keccak256("FINANCIAL_OWNER");
+bytes32 constant FINANCIAL_OPERATOR = keccak256("FINANCIAL_OPERATOR");
+
+/**
+ * @title GatewayAccessControl
+ * @author RIF Protocols Team
+ * @dev The main contract that that handles the role and their access within the gateway.
+ * This roles had been defined by the Security team and each address should from a multisig wallet.
+ */
+contract GatewayAccessControl is AccessControl, Ownable {
+    /**
+     * @notice Grants the default admin role OWNER to the deployer.
+     * @dev The default admin role is keccak256("OWNER"), which means that
+     * only accounts with this role will be able to grant or revoke
+     * other roles. By default, this role is also granted to the
+     * account that deploys the contract.
+     */
+    constructor() {
+        _setRoleAdmin(OWNER, OWNER);
```

Comment:
We need to see if we OWNER actually behaves like a role.

Reply:
Yes, I agree we could leave the functions described on the document for OWNER role to be performed by the owner of the contract, so we could remove OWNER role, but then we need to defined which role will grant and revoke roles, now only OWNER role can do so. Should we add an ADMIN role?